File: Selenium/Activities/Activity14.py
from selenium import webdriver
from selenium.webdriver.common.by import By
from selenium.webdriver.firefox.service import Service as FirefoxService
from webdriver_manager.firefox import GeckoDriverManager

service = FirefoxService(GeckoDriverManager().install())

# Start the Driver
with webdriver.Firefox(service=service) as driver:
    driver.get("https://v1.training-support.net/selenium/tables")
    print("Home title is: ", driver.title)

    cols = driver.find_elements(By.XPATH,"//table[@id='sortableTable']/thead/tr/th")
    print("Number of columns: " ,len(cols))

    rows = driver.find_elements(By.XPATH,"//table[contains(@id, 'sortableTable')]/tbody/tr")
    print("Number of rows: " ,len(rows));

    secondRowSecondData =  driver.find_element(By.XPATH,"//table[contains(@id,'sortableTable')]/tbody/tr[2]/td[2]")
    print("Second row second data: ",secondRowSecondData.text)

    driver.find_element(By.XPATH,"//table[@id='sortableTable']/thead/tr/th[1]").click()

    secondRowSecondData =  driver.find_element(By.XPATH,"//table[contains(@id,'sortableTable')]/tbody/tr[2]/td[2]")
    print("Second row second data: ",secondRowSecondData.text)

    footerRowData = driver.find_elements(By.XPATH,"//table[@id='sortableTable']/tfoot/tr/th")
    for cell in footerRowData:
        print("Footer Row Data: ",cell.text)


    driver.close();

   

File: Selenium/Activities/Activity14.java
//import org.openqa.selenium.WebDriver;
import org.openqa.selenium.*;
import org.openqa.selenium.firefox.FirefoxDriver;
import org.openqa.selenium.By;
import org.openqa.selenium.WebDriver;

import java.util.*;

public class Activity14 {
    public static void main(String[] args) {
        WebDriver driver = new FirefoxDriver();
        driver.get("https://v1.training-support.net/selenium/tables");
        System.out.println("Title : "+driver.getTitle());

        List<WebElement> cols = driver.findElements(By.xpath("//table[@id='sortableTable']/thead/tr/th"));
        System.out.println("Number of columns: " + cols.size());

        List<WebElement> rows = driver.findElements(By.xpath("//table[contains(@id, 'sortableTable')]/tbody/tr"));
        System.out.println("Number of rows: " + rows.size());

        WebElement secondRowSecondData =  driver.findElement(By.xpath("//table[contains(@id,'sortableTable')]/tbody/tr[2]/td[2]"));
        System.out.println("Second row second data: "+secondRowSecondData.getText());

        driver.findElement(By.xpath("//table[@id='sortableTable']/thead/tr/th[1]")).click();

        WebElement againSecondRowSecondData =  driver.findElement(By.xpath("//table[contains(@id,'sortableTable')]/tbody/tr[2]/td[2]"));
        System.out.println("Again Second row second data: "+againSecondRowSecondData.getText());

        List<WebElement> footer = driver.findElements(By.xpath("//table[@id='sortableTable']/tfoot/tr/th"));
        for(WebElement data:footer){
            System.out.println("Footer Data "+data.getText());
        }
       driver.close();
    }
}